crates/core/tedge_mapper/src/aws/mapper.rs

```diff
@@ -0,0 +1,66 @@
+use std::path::Path;
+
+use crate::{
+    aws::converter::AwsConverter,
+    core::{component::TEdgeComponent, mapper::create_mapper, size_threshold::SizeThreshold},
+};
+
+use async_trait::async_trait;
+use clock::WallClock;
+use tedge_config::{AzureMapperTimestamp, MqttBindAddressSetting, TEdgeConfig};
+use tedge_config::{ConfigSettingAccessor, MqttPortSetting};
+use tedge_utils::file::create_directory_with_user_group;
+use tracing::{info, info_span, Instrument};
+
+const AWS_MAPPER_NAME: &str = "tedge-mapper-aws";
+
+pub struct AwsMapper {}
+
+impl AwsMapper {
+    pub fn new() -> AwsMapper {
+        AwsMapper {}
+    }
+}
+
+#[async_trait]
+impl TEdgeComponent for AwsMapper {
+    fn session_name(&self) -> &str {
+        AWS_MAPPER_NAME
+    }
+
+    async fn init(&self, config_dir: &Path) -> Result<(), anyhow::Error> {
+        info!("Initialize tedge mapper aws");
+        create_directory_with_user_group(
+            format!("{}/operations/aws", config_dir.display()),
+            "tedge",
+            "tedge",
+            0o775,
+        )?;
+
+        self.init_session(AwsConverter::in_topic_filter()).await?;
+        Ok(())
+    }
+
+    async fn start(
+        &self,
+        tedge_config: TEdgeConfig,
+        _config_dir: &Path,
+    ) -> Result<(), anyhow::Error> {
+        let add_timestamp = tedge_config.query(AzureMapperTimestamp)?.is_set();
```

Comment:
This sounds unrelated to AWS.

Reply:
Thanks, added a timestamp config for AWS.